client/src/pages/OfferForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react'
+import './OfferForm.css'
+import { OfferData } from '../../types/offerTypes'
+
+interface OfferFormProps {
+  task: {
+    id: string
+    title: string
+  }
+  onSubmit: (offerData: OfferData) => void
+  onCancel: () => void
+}
+
+const OfferForm = ({ task, onSubmit, onCancel }: OfferFormProps) => {
+  const [offerData, setOfferData] = useState<OfferData>({
+    offer_description: '',
+    offer_media: null,
+    posting_time: '',
+    price: '',
+    offer_status: 'pending',
+  })
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setOfferData({
+      ...offerData,
+      [event.target.name]: event.target.value,
+    })
+  }
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0] || null
+    setOfferData({
+      ...offerData,
+      offer_media: file,
+    })
+  }
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const offerPayload = {
+      job_id: task.id,
+      worker_id: '1',
+      offer_description: offerData.offer_description,
+      posting_time: offerData.posting_time,
+      price: offerData.price,
+      offer_status: offerData.offer_status,
+      offer_media: offerData.offer_media,
+    }
+    onSubmit(offerPayload)
+  }
```

Comment:
I think using the previous state might be better here to avoid duplication. However, both are correct and have pros and cons.
```javascript
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const offerPayload = {
      ...offerData,
      job_id: task.id,
      worker_id: '1',
    };
    onSubmit(offerPayload);
  };
```

Reply:
I liked this one :)